src/components/Api.js

```diff
@@ -0,0 +1,147 @@
+export class Api {
+  constructor({ baseUrl, headers }) {
+    this._url = baseUrl;
+    this._authorization = headers.authorization;
+  }
+
+  getInitialCards() {
+    return fetch(`${this._url}/cards`, {
+      headers: {
+        authorization: this._authorization,
+        "Content-Type": "application/json",
+      },
+    })
+      .then((res) =>
+        res.ok
+          ? res.json()
+          : Promise.reject("An error has occurred", res.status)
+      )
```

Comment:
```js
.then(res => {
                if (res.ok) {
                    return res.json();
                }
                return Promise.reject(`Error ${res.status}`);
            })

```

This code is repeated in every request of the project. To avoid it you need to make a special method `_checkResponse` that will check any response from the server if it’s ok or not.


```js
_checkResponse(res) {
        // here is the code of the checking
    }

```

And now you can replace redundant repeating with the only line:


```js
 .then(this._checkResponse)

```

Please, pay attention that you need to pass only the `reference` for the method rather than the call. 

**Could be improved**

You can make a special method for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this._checkResponse)
  }

```

And now you can safely replace `fetch` with `this._request` and remove repeated checking. And you don’t need to change anything else.